### PR TITLE
WT-12456 Remove huffman encoding

### PR DESCRIPTION
**Pull request by korteland**
This removes the feature itself, replacing it with a stub that complains if the original table was configured with a non-empty config for `huffman_value`.

### CODE REVIEW

File: dist/api_data.py [hunk 1]
```diff
@@ -304,8 +304,7 @@ def __ge__(self, other):
     Config('huffman_key', 'none', r'''
         This option is no longer supported, retained for backward compatibility'''),
     Config('huffman_value', 'none', r'''
-        configure Huffman encoding for values. Permitted values are \c "none", \c "english",
-        \c "utf8<file>" or \c "utf16<file>". See @ref huffman for more information'''),
+        This option is no longer supported, retained for backward compatibility'''),
```
Comment (ruby-oujo) [inline dist/api_data.py:307]: I think we can add `undoc=True` if we no longer want to expose the API to future users.

Comment (korteland) [reply dist/api_data.py]: Good idea - done in e88e20d.

File: src/include/reconcile_inline.h [hunk 2]
```diff
@@ -417,19 +417,11 @@ __wt_rec_cell_build_val(WT_SESSION_IMPL *session, WT_RECONCILE *r, const void *d
...
-            WT_RET(__wt_huffman_encode(session, btree->huffman_value,
-              (const uint8_t *)val->buf.data, (uint32_t)val->buf.size, &val->buf));
-
-        /* Create an overflow object if the data won't fit. */
-        if (val->buf.size > btree->maxleafvalue) {
-            WT_STAT_CONN_DATA_INCR(session, rec_overflow_value);
-
-            return (__wt_rec_cell_build_ovfl(session, r, val, WT_CELL_VALUE_OVFL, tw, rle));
-        }
+    /* Create an overflow object if the data won't fit. */
```
[the remark this answers is not in the document]
Comment (sueloverso) [reply src/include/reconcile_inline.h:420]: Maybe add an assert. That feels like a safe assumption but if it isn't an assertion would catch it.

**Review by sueloverso — APPROVED** [review]
LGTM

File: src/include/reconcile_inline.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (korteland) [reply src/include/reconcile_inline.h:420]: Good call - done in 2786031.

File: dist/s_string.ok [hunk 3]
```diff
@@ -717,8 +713,6 @@ encodings
 encryptor
 encryptors
 endian
-english
-enqueue
```
Comment (ruby-oujo) [inline dist/s_string.ok:721]: honestly surprised some of these words aren't used elsewhere.

**Review by ruby-oujo — APPROVED** [review]
LGTM!

**Pull request merged**